tests/contrib/cherrypy/test_middleware.py

```diff
@@ -248,18 +208,14 @@ def test_unicode(self):
         s = spans[0]
         assert s.service == "test.cherrypy.service"
         assert s.resource == u"GET /üŋïĉóđē"
-        assert s.start >= start
-        assert s.duration <= end - start
         assert s.error == 0
         assert_span_http_status_code(s, 200)
         assert s.meta.get(http.METHOD) == "GET"
         assert s.meta.get(http.URL) == u"http://127.0.0.1:54583/üŋïĉóđē"
 
     def test_404(self):
-        start = time.time()
         self.getPage(u"/404/test")
-        time.sleep(0.01)  # Without this here, span may not be ready for inspection, and timings can be incorrect.
-        end = time.time()
+        time.sleep(0.1)
```

Comment:
do we still need those `time.sleep`? This is a really bad sign in general and proper to race condition. Is there any other way to make this working?

Reply:
Yeah it sucks. The way the CherryPy test helper works is by creating a subprocess to handle the requests. I tried to look for any kind of synchronization mechanism but there doesn't appear to be anything built into the helpers 😞

Reply:
😡